crates/solver/src/metrics.rs

```diff
@@ -226,14 +226,13 @@ impl Metrics {
         })?;
         registry.register(Box::new(complete_runloop_until_transaction.clone()))?;
 
-        let opts = prometheus::opts!(
-            "transaction_submission_seconds",
-            "Time it takes to submit a settlement transaction."
-        );
-        let transaction_submission = Histogram::with_opts(HistogramOpts {
-            common_opts: opts,
-            buckets: vec![f64::INFINITY],
-        })?;
+        let transaction_submission = HistogramVec::new(
```

Comment:
If we're mostly interested in the average time per strategy then in this case a 1 bucket histogram would be enough (only the +inf bucket). A one bucket histogram is a counter for the number of times the event happened and and a counter for the total duration duration measured. More buckets allow you to measure how the values are distributed but we might not need it here.

Like Martin said, we should manually set buckets.

Reply:
Added bucket definition. Is it fixed now?

Reply:
That's what I meant, yes. From the top of my head, I'm unsure if you need to define one INF bucket or zero buckets because the prometheus library adds it automatically. You could test it locally to see which one is right.